packages/grid/backend/grid/utils.py

```diff
@@ -29,8 +30,19 @@ def send_message_with_reply(
     if reply_to is None:
         reply_to = client.address
 
-    msg = message_type(address=address, reply_to=reply_to, **content)
-    reply = client.send_immediate_msg_with_reply(msg=msg)
+    use_new_service = flags.USE_NEW_SERVICE
+    if use_new_service:
+        msg = message_type(address=address, reply_to=reply_to, kwargs=content).sign(
+            signing_key=signing_key
+        )
+        reply = client.send_immediate_msg_with_reply(msg=msg)
+        try:
+            reply = reply.kwargs.upcast()
```

Comment:
Should this be using the `payload` getter instead of accessing kwargs directly? Perhaps we should make kwargs hidden so we can only access kwargs through the Request / Reply PyDantic models?

Reply:
fixed by : [90c8c73](https://github.com/OpenMined/PySyft/pull/6274/commits/90c8c739c4952e9238b3ab63d8d76eb3729a17d7)